### ISSUE TITLE
No sound card in HP Chromebook x360 14c-cc0013no

### ISSUE THREAD

**Issue by waterphotography** [thesofproject/linux#4110]
Latest chromiumos synced from main branch: repo init -u https://chromium.googlesource.com/chromiumos/manifest -b main

SOF: v1.9

Kernel:
1. the default v5.4 which is in the latest chromiumos main branch:
After board is set as volteer and chromiumos build is done, kernel image is updated into the device. No sound card could be found. 
After kernel log check, looks firmware's version is newer than kernel one.
[kernel_5_4.log](https://github.com/thesofproject/linux/files/10276585/kernel_5_4.log)

2. switch to v5.15 and sync/checkout the topic/sof-dev branch from github. 
After board is set as volteer and chromiumos build is done, kernel image is updated into the device. No sound card could be found.
After kernel log check, looks board config is not correct, as NHLT and hda codecs fail to be probed.
[kernel_5_15.log](https://github.com/thesofproject/linux/files/10276604/kernel_5_15.log)
Then modified the defconfig file of "chromiumos/src/third_party/chromiumos-overlay/eclass/cros-kernel/x86_64_defconfig" a bit, still no help. Maybe the modification is not quite correct.

3. Realized the kernel flavor is removed in v5.15 after the topic/sof-dev branch checkout. Then redo the steps. Before branch checkout, backup the chromeos flavor folder in the original v5.15 directory which is released together with the latest chromiumos. After the topic/sof-dev branch checkout, place back the backup chromeos flavor folder.
Then in the kernel build log, it looks it's using the flavor of chromeos-intel-pineview.
But then after this kernel image is updated into the device, even networks is not working in the target device, command line is not working either.

BTW, the base chromiumos build flashed in this target device is made from release in last year. Only kernel image is updated now.

Just would like to ask how to make the sound card working with the last codebase in this target device? Thank you!

**Comment by plbossart** [comment]
This doesn't seem to be an SOF issue proper, but rather a downstream Chrome-specific issue?
@cujomalainey FYI

**Comment by cujomalainey** [comment]
Yea we don't test our chromium code, so I would not be surprised if we have something wrong in the overlays that references something hidden. I'll take a look tomorrow

**Comment by cujomalainey** [comment]
@poloo5582 FYI

**Comment by waterphotography** [comment]
Updates:

1. I just now compiled and flashed the entire chromiumos build (with the default kernel v5.4) into the HP X360 device. Sound card can be registered.

2. I switched to kernel v5.15 and backed up the flavor folder of "chromeos" of the default v5.15 kernel. 

3. The topic/sof-dev branch from github was synced on kernel v5.15 and checked out. 

4. The "chromeos" flavor backup was restored back to the synced kernel v5.15 (topic/sof-dev branch).

5. Compile the new kernel v5.15 and update the kernel image into the device.

6. Sound card can be registered correctly. But the new kernel version in the new image is actually v6.1.

Just would like to double check, is it fine to reuse the flavor folder of "chromeos" of v5.15 kernel in the latest topic/sof-dev v6.1 kernel? Any risk?

Thanks!
